xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/renderer/xhtml/link/DocumentXHTMLLinkTypeRenderer.java

```diff
@@ -90,6 +103,31 @@ protected String computeLabel(ResourceReference reference)
         return this.linkLabelGenerator.generate(reference);
     }
 
+    /**
+     * Implementation for computing a document link title when no title has been specified.
+     * Looks for a component implementing URITitleGenerator with a role hint matching the reference scheme.
+     * @param reference the reference of the link for which to compute the label
+     * @return the computed title
+     * @since 15.2RC1
+     */
+    private String computeCreateTitle(ResourceReference reference)
```

Comment:
I think it is interesting to move the resolution of the `URITitleGenerator` resolution in a separate method, and to keep only the computation of the create title in this method.